packages/loot-design/src/components/ElementQuery.js

```diff
@@ -8,30 +13,23 @@ function findMatch(element, sizes) {
       (size.height != null && rect.height < size.height)
     );
   });
-
   return matched || sizes[sizes.length - 1];
 }
 
 // Component
+function ElementQuery({ children, sizes }) {
+  const elementRef = useRef(null);
+  const [matched, setMatched] = useState(null);
+  const observe = useResizeObserver(() => {
+    setMatched(findMatch(elementRef.current, sizes));
+  });
 
-function ElementQuery({ props: { children }, state: { matched }, inst }) {
-  return children(matched, el => (inst.element = el));
-}
+  useEffect(() => {
+    observe(elementRef.current);
+    return () => observe(null);
+  }, []);
 
-export default lively(ElementQuery, {
-  getInitialState() {
-    return { matched: null };
-  },
-  componentDidMount({ inst, props, setState }) {
-    inst.observer = new ResizeObserver(entries => {
```

Comment:
I didn't see a reason why we couldn't use the existing resizeObserver custom hook.